vignettes/summary_selecting.Rmd

```diff
@@ -0,0 +1,229 @@
+  ungroup()
+```
+
+
+## Formulas
+
+You can also use formulas to select variables:
+
+```{r}
+get_columns(iris, ~ Sepal.Length + Petal.Width)
+```
+
+This was mostly useful to use selection more easily in custom functions before
+`datawizard` 0.6.0, and is kept available for backward compatibility. 
+
+
+
+## Positions
+
+In addition to variable names, `select` can also take a numeric vector that indicates
+the position of the variables to select in the dataframe.
+
+```{r}
+get_columns(iris, c(1, 2, 5))
+```
+
+
+## Functions
+
+We can pass a function to `select`. This function will be applied to all columns
+and has to return `TRUE` or `FALSE`. For example, if we want to keep only numeric
+columns, we can use `is.numeric`.
+
+```{r}
+get_columns(iris, is.numeric)
+```
+
+Note that we can also define our own functions and pass them in `select`, 
+*as long as they return `TRUE` or `FALSE`* when applied to a column.
+
+```{r}
+my_function <- function(i) {
+  is.numeric(i) && mean(i, na.rm = TRUE) > 3.5
+}
+
+get_columns(iris, my_function)
+```
+
+
+## Patterns
+
+Most of the time, our data will be quite bigger than `iris`. It would be tedious
+to write the names of variables we want to select, and it would be fragile to rely
+on variable positions in the dataset as they may change later. In this case, we
+can use four select helpers: `starts_with()`, `ends_with()`, `contains()`, and
+`regex()`. The former three can take several patterns, while the latter takes
+a single regular expression.
+ 
+```{r}
+get_columns(iris, starts_with("Sep", "Peta"))
+
+get_columns(iris, ends_with("dth", "ies"))
+
+get_columns(iris, contains("pal", "ec"))
+
+get_columns(iris, regex("^Sep|ies"))
+```
+
+```{r echo=FALSE}
+row("Note: these functions are not exported by `datawizard` but are detected and
+applied internally. This means that they won't be detected by autocompletion
+when we write them.")
+```
+
+```{r echo=FALSE}
+row("Note #2: because these functions are not exported, they will not create 
+conflicts with the ones that come from the `tidyverse` and that have the same name.
+So we can still use `dplyr` and its friends, it won't change anything for selection
+in `datawizard` functions!")
+```
+
+
+# Excluding variables
+
+What if we want to keep all variables except for a few ones? There are two ways
+we can invert our selection.
+
+The first way is to put a minus sign `"-"` in front of the `select` argument.
+
+```{r}
+get_columns(iris, -c("Sepal.Length", "Petal.Width"))
+
+get_columns(iris, -starts_with("Sep", "Peta"))
+```
+
+Note that if we use numeric indices, we can't mix negative and positive values. 
+This means that we have to use `select = -(1:2)` if we want to exclude the first
+two columns, `select = -1:2` will not work:
+
+```{r}
+get_columns(iris, -(1:2))
+```
+
+The second way is to use the argument `exclude`. This argument has the same 
+possibilities as `select`. Note that we can use both `select` and `exclude` at 
+the same time.
```

Comment:
```suggestion
possibilities as `select`. Although this may not be required in most contexts, 
if we wanted to, we can use both `select` and `exclude` arguments at the same time.
```